app/controllers/api/v1/posts_controller.rb

```diff
@@ -20,12 +20,28 @@ def create
         @post.save if validate_post(@user, @post)
       end
 
+      def update
+        @post = Post.find(params[:id])
```

Comment:
You should create a before_action for this as you are also using it in the `show`